### DISCUSSION TITLE
Adapting to OPTForCausalLM models -- General/Academic Text completion for writing

### DISCUSSION THREAD

**Discussion by mbrotos** [moyix/fauxpilot#153] [Ideas]
tried forking the repo to run on the galactical model as an AI writing aid for academic uses. 

https://github.com/paperswithcode/galai

Been getting some weird results, likely due to tokenization any ideas?

Also just more generally, anyone know of a similar framework to fauxpilot for text completion more generally?

I think adding support for OPT or GPT models could be a step in this direction. Would appreciate any feedback or collaboration!

**Comment by thakkarparth007** [comment]
Hey @mbrotos you might find this useful: https://twitter.com/BlackHC/status/1612975716282368000 -- Andreas has integrated Galactica with Fauxpilot in a fork.

> Also just more generally, anyone know of a similar framework to fauxpilot for text completion more generally?

I'm not aware of something like this myself, except using a huggingface model directly. The problem with that is the inference will be really slow. Fauxpilot handles the speed issue by using FasterTransformer variants of the Codegen models, but FasterTransformer works with limited model families.
